### DISCUSSION TITLE
is setting /etc/locale ?

### DISCUSSION THREAD

**Discussion by ghost** [SpiralLinux/SpiralLinux-project#346] [Q&A] [answered]
Is setting /etc/locale the way to enforce locale, timezone, and choosen keyboard?  I am making an immutable environment for someone and I would like the whole thing reset back to ground zero with each reboot.

The user will just have access to the internet and local games but screwing with the system changes nothing unless she had access to the build envionment - which she doesnt.

I mean, she could technically install software but it will not stay, which is fine.

Thx/Bill

**Comment by geckolinux** [comment]
Hi there, `/etc/locale.conf` is only for the system language and time/date/currency regional formats. Notice that there is a bit of development churn relating to that in Debian, and the next SpiralLinux release will try to adapt to that a bit better: https://github.com/SpiralLinux/SpiralLinux-project/issues/326#issuecomment-1763448807

For the system keyboard it's `/etc/default/keyboard` , but notice that the user's desktop environment config can override that.

The system timezone is defined in `/etc/timezone` , and again the user's desktop environment config can override that.

**Reply by ghost** [reply]
yes, thank you, I totally overlooked the user environment.

I have done about 20 builds now with varying success.  My first issue was disk space, I made my build box with only 50GBs and it wasnt enough to do alot of builds, so I wiped it and reinstalled SpiralLinux with 250GB but it would not build anything - I discoved I had forgotten live-build and live-config, I rebuilt my build box with 75GB but noticed if you ever start a build but didnt change the 'user' it stays married to the 'live' user - build clean doesnt pickup later changes to the various hook files, mainly it doesnt pickup any changes to 2000-spiral under lib.

To get around this I have to wipe and reinstall the OS.  My current build box only has 8GB of RAM and no internal HDD.  So I dont really have the resources or bandwidth to also setup and troubleshoot why the VM may not be working, its just easier to reformat.

Is there any way I can force/trick the builder ti reread config changes?

Thx/Bill

**Reply by geckolinux** [reply, marked as answer]
Hi again,

Keep in mind that the hooks under `config/includes.chroot/lib/live/config/` get executed during the _live system boot process_, not during the ISO build process. So it sounds like you need to look into why that's not being executed during bootup. Are you sure the hook file is executable? Are the proper parameters present in the live boot kernel command line? Specifically, it needs to have `boot=live config` to make the previously mentioned hooks run.

You should never have to reinstall the host OS in order to apply config changes in a build. I prefer to use my `spiral` script (at the root of the tarballs that I upload) to automate the process a bit more, but even without that this will always work to start a new build with your current config and chroot includes:
```
lb clean
lb config
lb build

**Comment by ghost** [comment]
I need to step back from this, it does not pick up my changes.

Let me ask - will an ESP boot partition of 100MB be ok to share between SpiralLinux and say Windows 11?

I have to stop trying to set up my moms system and just live with the fact it will not survive with her using it ... so Tiny 11 may be my best course of action, it is fairly easy to just keep resetting that spin up when she breaks it  lol

I do appreciate your efforts tho.  I was a big time OpenSuSE fan but could not get into the Gecko spin.  I think SpiralLinux more my speed ... you know, snails and all  lol

No, Im just gonna take a break from trying to screw with the build process for now and build me a Win11 / Spiral monster

:) / Bill

**Reply by geckolinux** [reply]
> Let me ask - will an ESP boot partition of 100MB be ok to share between SpiralLinux and say Windows 11?

Hi there, don't confuse the partition that gets mounted as `/boot/efi/` on modern EFI systems with the `/boot/` partition that some old Linux distros used to recommend. My laptop with Linux and Windows has less than 30MB of data in the shared EFI partition, which Linux mounts to `/boot/efi/` . But the Linux kernels and initial RAM disk images are quite large and get placed in `/boot/` , which I would recommend being on the same `/` partition as the rest of the Linux system.

**Reply by ghost** [reply]
I didnt, I just keep wondering why M$ wants to stick with 100mb efi but then I remembered my laptop bios will just search for all efi and present a boot menu then I can pick an OS to boot  ... only gets nasty cuz Winders does not truly shutdown and leaves the disk in readonly mode  :(   But I can disable hibernate  lol

well, Im off to play with other Linux lunies  lol   I do have SpiralLinux and Tiny11 dual booting with Linux hidden under the ESP selection process  :)

**Reply by ghost** [reply]
I found out last year, as I was finishing my last build (before my mother-in-law died) that the EFI partition can be any size, even 2 TeraBytes - maybe larger.

I wanted to have local movies she could double click on and watch, using the EFI since it was already FAT32 was a perfect choice   lol

Bill
